nova/bigvm/manager.py

```diff
@@ -403,7 +403,7 @@ def _get_providers(self, context):
         # retrieve all bigvm provider's inventories
         for rp_uuid, rp in bigvm_providers.items():
             inventory = client._get_inventory(context, rp_uuid)
-            rp['inventory'] = inventory['inventories']
```

Comment:
Does the later code handle `inventory` properly? From a quick check, it looks to me like the code assumes `rp['inventory']` to always be a dictionary.